src/player.cpp

```diff
@@ -4121,24 +4121,24 @@ int player::overmap_sight_range(int light_level) const
     if( sight <= SEEX * 4) {
         return (sight / (SEEX / 2) );
     }
-    if ((has_amount("binoculars", 1) || has_amount("rifle_scope", 1) ||
-        has_amount("survivor_scope", 1) || -1 != weapon.has_gunmod("rifle_scope") ) &&
+    if ((has_item_with_flag("ZOOM") || worn_with_flag("ZOOM") || 
```

Comment:
The `worn_with_flag` is superfluous, `player::has_item_with_flag` calls `Character::has_item_with`, which checks the main inventory as well as worn and wielded items.

It also checks the contents of the items, so it might actually be enough to add the flag to the gunmod and be done (no special handling needed for an installed gunmod).